crate/server/src/core/operations/rekey.rs

```diff
@@ -0,0 +1,97 @@
+use cosmian_kmip::kmip::{
+    kmip_objects::ObjectType,
+    kmip_operations::{Create, ErrorReason, Import, ReKey, ReKeyResponse},
+    kmip_types::{StateEnumeration, UniqueIdentifier},
+};
+use cosmian_kms_client::access::ObjectOperationType;
+use tracing::{debug, trace};
+
+use crate::{
+    core::{
+        extra_database_params::ExtraDatabaseParams, operations::import::process_symmetric_key, KMS,
+    },
+    database::object_with_metadata::ObjectWithMetadata,
+    error::KmsError,
+    kms_bail,
+    result::{KResult, KResultHelper},
+};
+
+pub(crate) async fn rekey(
+    kms: &KMS,
+    request: ReKey,
+    owner: &str,
+    params: Option<&ExtraDatabaseParams>,
+) -> KResult<ReKeyResponse> {
+    trace!("ReKey: {}", serde_json::to_string(&request)?);
+
+    if request.protection_storage_masks.is_some() {
+        kms_bail!(KmsError::UnsupportedPlaceholder)
+    }
+
+    // there must be an identifier
```

Comment:
so it's not optionnal - cf. clap argument structure ?

Reply:
well It is optionnal in KMIP https://docs.oasis-open.org/kmip/kmip-spec/v2.0/os/kmip-spec-v2.0-os.html#_Toc6497568